noop.go

```diff
@@ -68,7 +70,25 @@ func (n NoopTracer) StartSpanWithOptions(opts StartSpanOptions) Span {
 	return defaultNoopSpan
 }
 
+// Extractor belongs to the Tracer interface.
+func (n NoopTracer) Extractor(format interface{}) Extractor {
+	return defaultNoopInjectorExtractor
+}
+
+// Injector belongs to the Tracer interface.
+func (n NoopTracer) Injector(format interface{}) Injector {
+	return defaultNoopInjectorExtractor
+}
+
 // JoinTrace belongs to the Tracer interface.
 func (n NoopTracer) JoinTrace(operationName string, parent interface{}) Span {
 	return defaultNoopSpan
 }
+
+func (n noopInjectorExtractor) InjectSpan(span Span, carrier interface{}) error {
+	return nil
+}
+
+func (n noopInjectorExtractor) JoinTrace(operationName string, carrier interface{}) (Span, error) {
```

Comment:
Shouldn't a nil return be used to signal that no trace could be found to
join with? Or should we reserve an error val for that?

On Friday, February 12, 2016, Yuri Shkuro <notifications@github.com> wrote:

> In noop.go
> <https://github.com/opentracing/opentracing-go/pull/48#discussion_r52794134>
> :
>
> >  // JoinTrace belongs to the Tracer interface.
> >  func (n NoopTracer) JoinTrace(operationName string, parent interface{}) Span {
> >  	return defaultNoopSpan
> >  }
> > +
> > +func (n noopInjectorExtractor) InjectSpan(span Span, carrier interface{}) error {
> > +	return nil
> > +}
> > +
> > +func (n noopInjectorExtractor) JoinTrace(operationName string, carrier interface{}) (Span, error) {
>
> it needs to return a noopSpan, otherwise end user code can NPE
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/opentracing/opentracing-go/pull/48/files#r52794134>.
>


-- 
(Sent via mobile)